lib/liquid/combinators/tags/comment.ex

```diff
@@ -0,0 +1,77 @@
+defmodule Liquid.Combinators.Tags.Comment do
+  @moduledoc """
+  Allows you to leave un-rendered code inside a Liquid template.
+  Any text within the opening and closing comment blocks will not be output,
+  and any Liquid code within will not be executed
+  Input:
+  ```
+    Anything you put between {% comment %} and {% endcomment %} tags
+    is turned into a comment.
+  ```
+  Output:
+  ```
+    Anything you put between  tags
+    is turned into a comment
+  ```
+  """
+  import NimbleParsec
+  alias Liquid.Combinators.{General, Tag, Markup}
+
+  @type t :: [comment: [String.t() | Comment.t() | Raw.t()]]
+
+  def comment_content do
+    General.literal_until_tag()
+    |> optional(
+      choice([
+        parsec(:comment) |> optional(parsec(:comment_content)),
+        parsec(:raw) |> optional(parsec(:comment_content)),
+        any_tag() |> optional(parsec(:comment_content))
+      ])
+    )
+    |> concat(General.literal_until_tag())
+  end
+
+  def tag do
+    Tag.define_closed("comment", & &1, fn combinator ->
+      combinator
+      |> optional(parsec(:comment_content))
+      |> reduce({Markup, :literal, []})
+    end)
+  end
+
+  def any_tag do
+    empty()
+    |> string(General.codepoints().start_tag)
+    |> optional(repeat(General.whitespace()))
+    |> choice([
+      string_with_comment(),
+      string_with_endcomment(),
+      string_without_comment()
+    ])
+    |> reduce({List, :to_string, []})
```

Comment:
why is the parsed AST being converted to a string?

Reply:
Parser works in a binary fashion, char by char. The result is returned like a list of codepoints and we must transform it into a string.